runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -222,14 +218,42 @@ public boolean mergeSegments(Layout currentLayout)
             newLayout = currentLayout;
         }
         reconfigureSequencerServers(currentLayout, newLayout, false);
-        return true;
+    }
+
+    /**
+     * Add the log unit to the segment to increase redundancy. This is preceded by state transfer.
+     *
+     * @param currentLayout Current layout.
+     * @param endpoint      Endpoint to be added to the segment.
+     * @param stripeIndex   Stripe index.
+     * @throws OutrankedException if consensus is outranked.
+     */
+    public void addLogUnitReplica(@Nonnull Layout currentLayout,
+                                  @Nonnull String endpoint,
+                                  int stripeIndex) throws OutrankedException {
+        boolean isTaskDone = currentLayout.getSegments().stream()
```

Comment:
needs more comment as these methods are too verbose and method names too layout specific to convey the intent.

Reply:
Done.